src/views/scans/showScansModal.tsx

```diff
@@ -27,14 +27,40 @@ interface ShowScansModalProps {
 const ShowScansModal: React.FC<ShowScansModalProps> = ({
   isOpen,
   scan,
-  scanJobs,
+  scanJobs = [],
   onDownload = Function.prototype,
   onClose = Function.prototype,
   actions
 }) => {
   const { t } = useTranslation();
   const [activeSortIndex, setActiveSortIndex] = useState<number | undefined>();
   const [activeSortDirection, setActiveSortDirection] = useState<'asc' | 'desc' | undefined>();
+  const [sortedScanJobs, setSortedScanJobs] = useState(scanJobs);
+
+  useEffect(() => {
+    setSortedScanJobs(scanJobs);
+  }, [scanJobs]);
+
+  const getSortValue = (job: Pick<ScanJobType, 'start_time' | 'end_time' | 'status'>, index: number) => {
+    if (index === 0) {
+      return new Date(job.end_time || job.start_time).getTime();
+    }
+    if (index === 1) {
+      return job.status;
+    }
+    return '';
+  };
+
+  const sortJobs = (jobs: typeof scanJobs, index: number, direction: 'asc' | 'desc') => {
```

Comment:
recommended:
you should consider breaking these helpers out into independent helper functions for testing